### DISCUSSION TITLE
umi4 useModel如果返回改变state的方法，为什么必须要用useCallback包裹，经测试如果不用callback包裹，在使用的地方通过useModel第二个参数返回这个方法

### DISCUSSION THREAD

**Discussion by buquan** [umijs/umi#12271] [General]
<img width="959" alt="image" src="https://github.com/umijs/umi/assets/23711051/8eef8275-b9b0-4f58-8244-37498d328cda">

**Comment by fz6m** [comment]
你是指为什么要造一个 `increment` 函数吗，在 hooks 里面（如 react 组件或 `useXXX` 的 hooks 函数内），函数如果不用 `useCallback` 优化会在每次渲染时都重新创建，此处没有任何 deps 依赖就用 `useCallback` 优化了，是没什么问题的。

但是否应该用 `useCallback` ，什么时候用，在哪些地方用，用多少，为什么用，为什么不用，关于此处更多内容非常深奥，请自行学习。
